(GH-49) feat :directions for buildings

ryad-all opened the pull request:
# Objectives of the Pull Request

This pull request aims to:

- **Add a new feature for in-map directions:**  
  Add new tests for the in-app “Directions” functionality. These tests ensure that:
  - When both start and destination buildings are set, the "Directions" button appears.
  - Pressing "Directions" fetches and displays a dotted route with turn-by-turn instructions.
  - The "Cancel Navigation" button clears the route and instructions.

- **Improve Overall Code Stability:**  
  Introduce safe type assertions and proper Jest configuration to prevent issues (e.g., with `global.fetch`), ensuring the test environment works consistently.

# How to Run the Pull Request

```bash
# Navigate to the project directory containing the "minicap/application" folder.
git checkout feat/directions_for_buildings
git pull


# Install dependencies:
npm install

# Install dependencies 
npm install --save-dev metro-react-native-babel-preset
npm install react-native-dotenv
npx expo install jest-expo
npm i

# Start the application with Expo:
npx expo start

# To run tests and update snapshots:
npm test -- -u

ryad-all committed:
(GH-49) feat/added directions on the campus map(please paste your own google API key)

ryad-all committed:
(GH-49) fix: Align React & react-test-renderer versions to 18.2.0

it had issue aligning and rendering components from different versions of react

ryad-all committed:
(GH-49) test: add tests for campus map directions functionality

- Add tests to verify that the "Directions" button appears when both start and destination buildings are set.
- Verify that in-app directions are fetched and rendered correctly.
- Ensure that "Cancel Navigation" clears the route and directions.

ryad-all committed:
Update ci-pipeline.yml

ryad-all committed:
(GH-49) fix: node modules installed

ryad-all committed:
Merge branch 'feat/directions_for_buildings' of https://github.com/AlimuratDinch/minicap into feat/directions_for_buildings

AlimuratDinch commented:
@ryad-all the name of the Pull Request should be like this `(GH-49) feat :directions for buildings`

Questions: 
Why we need to push `node modules` and `.expo` folder?

ryad-all commented:
> @ryad-all the name of the Pull Request should be like this `(GH-49) feat :directions for buildings`
> 
> Questions: Why we need to push `node modules` and `.expo` folder?

Actually no need to push, just make sure you have the correct React version and re-install npm packages with the command "npm install"

ryad-all committed:
(GH-49) fix: support legacy coordinate format in parser

ryad-all committed:
(GH-49) fix: node modules causing security issues

ryad-all committed:
(GH-49) fix: security issue with regex fixed

bulabd commented:
I followed the commands specified and got this error:
![{4BDCA881-5F94-41F0-A130-DC4BB04506BB}](https://github.com/user-attachments/assets/018f4245-6a75-4d94-a1bf-13e61aff2219)
Have you gotten this error before?

ryad-all commented:
> I followed the commands specified and got this error: ![{4BDCA881-5F94-41F0-A130-DC4BB04506BB}](https://private-user-images.githubusercontent.com/98482265/416048240-018f4245-6a75-4d94-a1bf-13e61aff2219.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDAzNjM5MjUsIm5iZiI6MTc0MDM2MzYyNSwicGF0aCI6Ii85ODQ4MjI2NS80MTYwNDgyNDAtMDE4ZjQyNDUtNmE3NS00ZDk0LWExYmYtMTNlNjFhZmYyMjE5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjI0VDAyMjAyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMyODdkMDc2OWY1MTA5ZmJjODM0NWVmMzY1M2FkY2VjOTlhMDViNWM3YjgxMzk3NWM1Y2ZlZDI5YTY5YTY2ZGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.7itjtx7Y0GhnLvDRX1XshVw0eFYnOWC4_69J6o3O-CE) Have you gotten this error before?

did u run npm install?

bulabd commented:
Yes I did. It installed the first time but with a lot of warnings. I tried to run and got the error above. Tried to do npm install again and it crashed also.

ryad-all commented:
> Yes I did. It installed the first time but with a lot of warnings. I tried to run and got the error above. Tried to do npm install again and it crashed also.

try this: npm ci --legacy-peer-deps

ryad-all commented:
> It worked thanks alot! Unfortunately I get this error: ![{35AC78E5-6F8C-4733-855C-559BFEFF0B99}](https://private-user-images.githubusercontent.com/98482265/416050059-e10d0ffa-bf84-4eb0-a4bd-a65cdf784d0f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDAzNjQ4MTYsIm5iZiI6MTc0MDM2NDUxNiwicGF0aCI6Ii85ODQ4MjI2NS80MTYwNTAwNTktZTEwZDBmZmEtYmY4NC00ZWIwLWE0YmQtYTY1Y2RmNzg0ZDBmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjI0VDAyMzUxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ1NDM0OWUzMTI4NzIxZGYwYWIwNTRmMDU2ZDAxNGVlZjY0ZTM5ZGExOTI2NGEzYjM3ZDY5ODNjZTYzYzhkNjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.y7imvNDUJUr_egMbJcpuZNVfGFFyPL7SnRB5Fadwlnc)

its because u need to put ur api key here in the code, i didnt push it for security reasons.
![image](https://github.com/user-attachments/assets/8657dd6e-8060-4388-9ee3-712f9e9bf6dd)
So make sure to put your API key

bulabd commented:
![{F7AF0708-288F-44B4-A29E-D8BBBEF72A08}](https://github.com/user-attachments/assets/291a9e98-7c62-4694-838f-cbdaa7c166b8)
Very nice and works!
I'll approve once you set up so that it uses the API from the .env file.

ryad-all committed:
(GH-49) fix: inhanced securtiy by gettiing the api key from .env file

AlimuratDinch commented:
![image](https://github.com/user-attachments/assets/c652b527-7b56-481a-b174-3aa95ad66b46)
Do the tests pass for you?

AlimuratDinch reviewed (changes requested):
i am getting this error
![image](https://github.com/user-attachments/assets/0df353f5-fc6d-4146-902c-3f97e3c870cd)

AlimuratDinch commented:
I think we should have more approvals for this PR.

ryad-all committed:
(GH-49) feat: added map directions

ryad-all committed:
Merge branch 'feat/onMap-directions' into feat/directions_for_buildings

ryad-all committed:
Update ci-pipeline.yml

ryad-all committed:
(GH-49) fix: fixed @env export

ryad-all committed:
Merge branch 'feat/directions_for_buildings' of https://github.com/AlimuratDinch/minicap into feat/directions_for_buildings

ryad-all committed:
(GH-49) fix: added @env to be recognised in checks

ryad-all committed:
(GH-49) test: enhanced tests to add @env

ryad-all committed:
(GH-49) test: fixed issue with API_KEY

- Added global mock for @env in Jest setup to provide a dummy API_KEY.
- Updated jest.config.js to include the setup file.
- Fixed tests to no longer fail due to missing API_KEY.

ryad-all committed:
(GH-49): fix: Add global declaration for jest in jest.setup.js

ryad-all commented:
> i am getting this error ![image](https://private-user-images.githubusercontent.com/132238708/416227003-0df353f5-fc6d-4146-902c-3f97e3c870cd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDA0MjUxNzcsIm5iZiI6MTc0MDQyNDg3NywicGF0aCI6Ii8xMzIyMzg3MDgvNDE2MjI3MDAzLTBkZjM1M2Y1LWZjNmQtNDE0Ni05MDJjLTNmOTdlM2M4NzBjZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjI0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIyNFQxOTIxMTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iMTMxZmZhN2RjNjg3OTE5ZTZjZTIyMGUzMGFlNTFhNWZjY2RhZTBkYWIzODU0NGI1NTEyOTgzZmIzZGFmNjllJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.3jk3CoxQx3uX30wRuef5imbE-RprgCWO2qBkxTl2_u8)

I have now made the changes and updated the packages. Also,  I used the google api key directly from the environment for security reasons. Therefore, no need to paste it in the code.

Summary of my changes:
- Upgraded various packages to the recommended versions for optimal Expo compatibility.
- Updated dependencies and configuration (including Babel and Jest) to resolve environment variable issues.
- Added a global Jest mock for @env to supply a dummy API_KEY in tests.
- Verified that all tests pass after these updates.

AlimuratDinch commented:
@ryad-all I am getting this issue
![image](https://github.com/user-attachments/assets/beb90d80-643c-44a4-a213-5f95def305e9)

Sawanoza commented:
I am also getting a similar error.
![image](https://github.com/user-attachments/assets/d3ad9f2d-2f9b-4546-acb6-ad433ff01bfc)

NekZampe commented:
I am getting this issue: 
![image](https://github.com/user-attachments/assets/de4a7170-090a-406a-9d44-0474b1eb50b3)

NekZampe commented:
> I am getting this issue: ![image](https://private-user-images.githubusercontent.com/127551641/416965638-de4a7170-090a-406a-9d44-0474b1eb50b3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDA1NDMwNjcsIm5iZiI6MTc0MDU0Mjc2NywicGF0aCI6Ii8xMjc1NTE2NDEvNDE2OTY1NjM4LWRlNGE3MTcwLTA5MGEtNDA2YS05ZDQ0LTA0NzRiMWViNTBiMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIyNlQwNDA2MDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03OGRjOGVkN2ZmNzNlOTllNzhkMzhiOTlmZjE4MmMzYTY1MWJkZmZiNmRiZTI3MjIwMzdhMDgwMWIwZGZkNTc1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.I79B9UYDvs254ctGDbPdPxZpgmrD2pDa0uT6pyQSnEg)
> 
> but when I run the tests they all pass: ![image](https://private-user-images.githubusercontent.com/127551641/416966055-5393f990-30f5-422a-8672-a6f264eb8365.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDA1NDMwNjcsIm5iZiI6MTc0MDU0Mjc2NywicGF0aCI6Ii8xMjc1NTE2NDEvNDE2OTY2MDU1LTUzOTNmOTkwLTMwZjUtNDIyYS04NjcyLWE2ZjI2NGViODM2NS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIyNlQwNDA2MDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iYmYzN2UxMjMyOTI2Nzg2N2E1YWIyNDUyMDE4OTkyYWE1YjM1ZjZkNzJjZDU2MWM0ODg5YmI4ODA4ZGFlYTU5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.f1QM5-5Uwq83Qwx8C8gd27LChBAtmIJtJABoEzFyeXQ)

I was able to fix the first issue by changing the babbel.config.js file and removing all presets but "babel-preset-expo" but a new one came up, If i try to navigate from one building to another i get: 
![image](https://github.com/user-attachments/assets/f5423117-6043-42b4-9295-61bd67c02efc)

ryad-all commented:
I'm currently investigating what went wrong. I want to note that commit 85be92c71142e05b5a4640becaf24752a024adc7 has been successfully run by a teammate, so the issue seems isolated to my environment/configuration.

For security reasons, I must integrate the API key via a .env file; otherwise, the API key would be hardcoded in the codebase, which is not acceptable. I'm working on finding a solution that meets both our functionality and security requirements.

I'll update the PR once I have a fix.

ryad-all committed:
(GH-49) fix: fixed some issues with the packages
